lib/actions/reassemble.js

```diff
@@ -54,19 +87,196 @@ async function processAction(msg) {
     Currently the group has ${messagesNumberSeen} of ${messageGroupSize} message(s).`,
   );
 
-  if (messagesNumberSeen >= messageGroupSize) {
-    parsedMessageGroup.messages.forEach((message) => {
-      incomingData[message.messageId] = message.messageData;
-    });
+  // when grouSized option is selected
+  if (mode === 'groupSize') {
+    if (messagesNumberSeen >= messageGroupSize) {
+      parsedMessageGroup.messages.forEach((message) => {
+        incomingData[message.messageId] = message.messageData;
+      });
+      await this.emit('data', messages.newMessageWithBody({
+        groupSize,
+        groupId,
+        messageData: incomingData,
+      }));
+      await storage.deleteObjectById(messageGroupId);
+      this.logger.info(`Message group with id ${messageGroupId} has been deleted`);
+    }
+  }
 
-    await this.emit('data', messages.newMessageWithBody({
-      groupSize,
-      groupId,
-      messageData: incomingData,
-    }));
-    await storage.deleteObjectById(messageGroupId);
-    this.logger.info(`Message group with id ${messageGroupId} has been deleted`);
+  // When delay timer option is selected
+  if (mode === 'timeout') {
+    delay = msg.body.timersec;
+    delay = (msg.body.timersec >= 40000) ? 40000 : msg.body.timersec;
+    clearTimeout(timeHandle);
+    timeHandle = setTimeout(timer, delay, this);
+    if (groupList.includes(messageGroupId)) {
+      // no action required
+    } else {
+      groupList.push(messageGroupId);
+    }
+  }
+
+  // When both groupSize and delay timer option is selected
+  if (mode === 'groupSize&timeout') {
+    delay = msg.body.timersec;
+    delay = (msg.body.timersec >= 40000) ? 40000 : msg.body.timersec;
```

Comment:
Duplicate assingment.

Reply:
They are been used in 2 different code branch
'timeout' & 'groupSize&timeout'

Reply:
I'm saying that assigning the value of `msg.body.timersec` to `delay` is pointless if `delay` is immediately reassigned to `(msg.body.timersec >= 40000) ? 40000 : msg.body.timersec;`